readthedocs/doc_builder/backends/mkdocs.py

```diff
@@ -327,27 +327,60 @@ class MkdocsHTML(BaseMkdocs):
     build_dir = '_build/html'
 
 
-class SafeLoaderIgnoreUnknown(yaml.SafeLoader):  # pylint: disable=too-many-ancestors
+class ProxyPythonName(yaml.YAMLObject):
+    def __init__(self, value):
+        self.value = value
+
+
+class SafeLoader(yaml.SafeLoader):  # pylint: disable=too-many-ancestors
 
     """
-    YAML loader to ignore unknown tags.
+    Safe YAML loader.
 
-    Borrowed from https://stackoverflow.com/a/57121993
+    This loader parses special ``!!python/name:`` tags without actually
+    importing or executing code. Every other special tag is ignored.
+
+    Borrowed from https://stackoverflow.com/a/57121993.
+    Issue https://github.com/readthedocs/readthedocs.org/issues/7461.
     """
 
     def ignore_unknown(self, node):  # pylint: disable=no-self-use, unused-argument
         return None
 
+    def construct_python_name(self, suffix, node):  # pylint: disable=no-self-use, unused-argument
```

Comment:
Shouldn't we need the same for others like: `construct_python_module`, `construct_python_object` and maybe others here as well?

https://github.com/yaml/pyyaml/blob/538b5c93f7d5dee40322893c1e524e94a4f8bbde/lib3/yaml/constructor.py#L572

Reply:
If you think we should immediately add support for all `python/*` tags, then yes. I only added logic for the `python/name` one since it's the only one I use, and the only one that is mentioned in the original issue https://github.com/readthedocs/readthedocs.org/issues/6889.

Just tell me and I'll update the PR to add support for the other tags!

Reply:
I think it depends on if it's common to have those... But I'd say it doesn't hurt to add them now

Reply:
There are 16 more Python tags:

tag | python
-- | --
!!python/none | None
!!python/bool | bool
!!python/bytes | (bytes in Python 3)
!!python/str | str (str in Python 3)
!!python/unicode | unicode (str in Python 3)
!!python/int | int
!!python/long | long (int in Python 3)
!!python/float | float
!!python/complex | complex
!!python/list | list
!!python/tuple | tuple
!!python/dict | dict
!!python/name:module.name | module.name
!!python/module:package.module | package.module
!!python/object:module.cls | module.cls instance
!!python/object/new:module.cls | module.cls instance
!!python/object/apply:module.f | value of f(...)

Taken from https://pyyaml.org/wiki/PyYAMLDocumentation#yaml-tags-and-python-types.

Honestly I'm a bit reluctant to add all these, especially at once. Some of these tags might be more complex to support (more than one suffix/value to store in a proxy object, multiple ways to write them, etc.). I think it would be best to add them one by one as people ask for them. We now know how, so it could be quick PRs.

Reply:
I think it makes sense to add only what's required now to make MkDocs with plugins to work :+1: